### PR TITLE
Adding analytics to docs site

### PR DESCRIPTION
**Pull request by dbanksdesign**
*Issue #, if available:*

*Description of changes:* Adding in the tracking code from the docs repo with some tweaks. I can see the analytics requests going out for each page visit:

CleanShot 2021-12-13 at 16 01 50@2x

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/src/pages/_app.page.tsx [hunk 1]
```diff
@@ -19,6 +20,9 @@ function MyApp({ Component, pageProps }) {
   const [colorMode, setColorMode] = React.useState<ColorMode>('system');
   const [themeOverride, setThemeOverride] = React.useState('');
 
+  configure();
```
Comment (dbanksdesign) [inline docs/src/pages/_app.page.tsx:23]: This has a check to see if it is configured yet, so running it for each page render is ok?

File: docs/src/pages/_document.page.tsx [hunk 2]
```diff
@@ -25,16 +25,20 @@ const getCSPContent = (context: Readonly<HtmlProps>) => {
       style-src 'self' 'unsafe-inline';
       font-src 'self' data:;
       frame-src *.codesandbox.io;
-      script-src 'unsafe-eval' 'self' '${cspInlineScriptHash}'
+      img-src 'self' cm.everesttech.net amazonwebservices.d2.sc.omtrdc.net;
+      connect-src 'self' *.shortbread.aws.dev dpm.demdex.net;
+      script-src 'unsafe-eval' 'self' '${cspInlineScriptHash}' a0.awsstatic.com;
```
Comment (dbanksdesign) [inline docs/src/pages/_document.page.tsx:30]: These are all what the tracking stuff needs. I figured these out by building, seeing an error, adding the domain to the proper place, repeat.

**Review by dbanksdesign — COMMENTED** [review]
Take a look around.

File: docs/src/utils/track.ts [hunk 3]
```diff
@@ -0,0 +1,161 @@
+// https://github.com/aws-amplify/docs/blob/main/src/utils/track.ts
+let configured = false;
+
+declare global {
+  interface Window {
+    s: any;
```
Comment (hvergara) [inline docs/src/utils/track.ts:6]: can we use a different variable name? `s` seems to be a little bit confusing

**Review by hvergara — APPROVED** [review]

File: docs/src/pages/_document.page.tsx [hunk 4]
```diff
@@ -25,16 +25,20 @@ const getCSPContent = (context: Readonly<HtmlProps>) => {
       style-src 'self' 'unsafe-inline';
       font-src 'self' data:;
       frame-src *.codesandbox.io;
-      script-src 'unsafe-eval' 'self' '${cspInlineScriptHash}'
+      img-src 'self' cm.everesttech.net amazonwebservices.d2.sc.omtrdc.net;
```
Comment (reesscot) [inline docs/src/pages/_document.page.tsx:28]: Do you want to add `dpm.demdex.net`; here as well?
```suggestion
      img-src 'self' cm.everesttech.net amazonwebservices.d2.sc.omtrdc.net dpm.demdex.net;
```

File: docs/src/utils/track.ts (hunk 3 above)
Comment (reesscot) [reply docs/src/utils/track.ts:6]: Pretty sure `s` is what Adobe analytics requires...

**Review by reesscot — APPROVED** [review]

File: docs/src/pages/_app.page.tsx [hunk 5]
```diff
@@ -19,6 +20,9 @@ function MyApp({ Component, pageProps }) {
...
+  trackPageVisit();
```
[the remark this answers is not in the document]
Comment (reesscot) [reply docs/src/pages/_app.page.tsx:24]: Should we wrap this in a conditional to only run in prod mode?

File: docs/src/utils/track.ts (hunk 3 above)
Comment (dbanksdesign) [reply docs/src/utils/track.ts:6]: Yup that variable is not set by us

**Pull request merged**